src/typecheck.rs

```diff
@@ -930,3 +1013,18 @@ fn tc_memarg(context: &mut Context, params: &mut [ast::Expression], span: &Span)
 
     Ok(())
 }
+
+fn tc_lane(context: &mut Context, param: &mut ast::Expression, span: &Span) -> Result<()> {
+    if param.type_ != Some(I32) {
+        return type_mismatch(Some(I32), &span, param.type_, &param.span, context.sources);
+    }
+    let lane = param.const_i32();
+    if lane < 0 || lane > 15 {
```

Comment:
This definitely needs to check for the correct range for this specific instruction. The emit phase will most likely just panic when encountering an out-of-range lane.

But let me read the spec first, so that I can give a suggestion on how to best implement that.

Reply:
Ahh, okay, that makes sense. It looked like there was a fairly simple way to add these checks so I've tried to do that, but I'm certainly open to alternatives!